### PR TITLE
Update libyaml_vendor to 0.2.5.

### PR DESCRIPTION
**Pull request by clalancette**
Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

### CODE REVIEW

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14015/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8837/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11709/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14099/)

**Comment by clalancette** [comment]
Windows Debug is en fuego for other reasons, so shouldn't hold this one up.

File: CMakeLists.txt [hunk 1]
```diff
@@ -60,9 +60,11 @@ macro(build_libyaml)
     endif()
   endif()
   include(ExternalProject)
-  externalproject_add(libyaml-10c9078
-    URL https://github.com/yaml/libyaml/archive/10c907871f1ccd779c7fccf6b81a62762b5c4e7b.zip
-    URL_MD5 b595f0ab0735c04c7f6ed0a798a3eff7
+  externalproject_add(libyaml-0.2.5
+    GIT_REPOSITORY https://github.com/yaml/libyaml
+    GIT_TAG 0.2.5
+    GIT_CONFIG advice.detachedHead=false
```
Comment (clalancette) [inline CMakeLists.txt:66]: Note that I switched to git here because the latest tarball that is released is very broken w.r.t. CMake.

Comment (wjwwood) [reply CMakeLists.txt:66]: I'm a bit confused why the tarball is broken but the tag is ok?

I mean we could use https://github.com/yaml/libyaml/archive/refs/tags/0.2.5.zip right?

I don't know if it matters, I just don't know why we need git to get the code.

Comment (clalancette) [reply CMakeLists.txt:66]: > I'm a bit confused why the tarball is broken but the tag is ok?

I don't really know why.  But if you pull https://github.com/yaml/libyaml/releases/download/0.2.5/yaml-0.2.5.tar.gz , it is missing crucial pieces of the cmake infrastructure (like the file `tests/CMakeLists.txt`).

However, you are correct that the zip file *does* contain all of the necessary files.  So I'll switch back to that, thanks for pointing it out.

**Comment by clalancette** [comment]
New CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14113/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8921/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11802/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14209/)

**Review by wjwwood — APPROVED** [review]

**Comment by clalancette** [comment]
The warnings on Windows and macOS are not because of this PR.  So going ahead and merging, thanks for the review.

**Pull request merged**